moban/utils.py

```diff
@@ -133,30 +133,27 @@ def pip_install(packages):
 
 
 def git_clone(repos, submodule=False):
-    import subprocess
+    from git import Repo
 
     moban_home = get_moban_home()
     mkdir_p(moban_home)
 
     for repo in repos:
         repo_name = get_repo_name(repo)
         local_repo_folder = os.path.join(moban_home, repo_name)
-        current_working_dir = os.getcwd()
         if os.path.exists(local_repo_folder):
             reporter.report_git_pull(repo_name)
-            os.chdir(local_repo_folder)
-            subprocess.check_call(["git", "pull"])
+            repo = Repo(local_repo_folder)
+            repo.git.pull()
             if submodule:
-                subprocess.check_call(["git", "submodule", "update"])
+                reporter.report_info_message("updating submodule")
+                repo.git.submodule("update")
         else:
             reporter.report_git_clone(repo_name)
-            os.chdir(moban_home)
-            subprocess.check_call(["git", "clone", repo, repo_name])
+            repo = Repo.clone_from(repo, local_repo_folder)
             if submodule:
-                os.chdir(os.path.join(moban_home, repo_name))
-                subprocess.check_call(["git", "submodule", "init"])
-                subprocess.check_call(["git", "submodule", "update"])
-        os.chdir(current_working_dir)
+                reporter.report_info_message("checking out submodule")
+                repo.git.submodule("update", "--init")
```

Comment:
We should test this on vm without git installed, to be sure it doesnt depend on git

Maybe add a new git script in PATH on travis which just does `exit 2`.

There might be a way to avoid this , esp. if .gitmodules doesnt exist.

Reply:
GitPython *should* default to a pure python solution, so `git` cli is unnecessary for most operations.  but this `git submodules` worries me - I suspect there is no pure python implementation of it.

Reply:
https://github.com/moremoban/moban/issues/207 ssh not working

Reply:
http private works.  But it is using subprocesses.  Anyways, this is definately better.

Reply:
Despite this, there are likely some operations which will be able to be done without `git-core`, especially any git db operations, so there will be less subprocesses, and/or we can do more intensive git queries on existing repos after they are cloned without incurring more subprocess costs.

Reply:
moban can alert user when git is indeed used, which is much more friendly. will include it in another PR.